doc/user/content/sql/create-source/kafka.md

```diff
@@ -0,0 +1,372 @@
+
+- This option requires specifying the key and value encodings explicitly using the `KEY FORMAT ... VALUE FORMAT` [syntax](#syntax).
+
+- The `UPSERT` envelope always includes keys.
+
+- The `DEBEZIUM` envelope is incompatible with this option.
+
+#### Partition, offset, timestamp
+
+These metadata fields are exposed via the `INCLUDE PARTITION`, `INCLUDE OFFSET` and `INCLUDE TIMESTAMP` options.
+
+```sql
+CREATE SOURCE kafka_metadata
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://localhost:8081'
+  INCLUDE PARTITION, OFFSET, TIMESTAMP AS ts
+  ENVELOPE NONE;
+```
+
+```sql
+SELECT "offset" FROM kafka_metadata WHERE ts > '2021-01-01';
+
+offset
+------
+15
+14
+13
+```
+
+Note that:
+
+- Using the `INCLUDE OFFSET` option with Debezium requires `UPSERT` semantics.
+
+### Setting start offsets
+
+To start consuming a Kafka stream from a specific offset, you can use the `start_offset` option.
+
+```sql
+CREATE MATERIALIZED SOURCE kafka_offset
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
+  -- Start reading from the earliest offset in the first partition,
+  -- the second partition at 10, and the third partition at 100
+  WITH (start_offset=[0,10,100])
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://localhost:8081';
+```
+
+Note that:
+
+- If fewer offsets than partitions are provided, the remaining partitions will start at offset 0. This is true if you provide `start_offset=1` or `start_offset=[1, ...]`.
+- If more offsets than partitions are provided, then any partitions added later will incorrectly be read from that offset. So, if you have a single partition, but you provide `start_offset=[1,2]`, when you add the second partition you will miss the first 2 records of data.
+- Using an offset with a source envelope that can supply updates or deletes requires that Materialize handle possibly nonsensical events (e.g. an update for a row that was never inserted). For that reason, starting at an offset requires either a `NONE` envelope or a `(DEBEZIUM) UPSERT` envelope.
+
+#### Time-based offsets
+
+It's also possible to set a start offset based on Kafka timestamps, using the `kafka_time_offset` option. This approach sets the start offset for each available partition based on the Kafka timestamp and the source behaves as if `start_offset` was provided directly.
+
+{{< note >}}
+The `kafka_time_offset` option is not supported yet for Redpanda sources ([Redpanda #2397](https://github.com/vectorizedio/redpanda/issues/2397)).
+{{</ note >}}
+
+It's important to note that `kafka_time_offset` is a property of the source: it will be calculated _once_ at the time the `CREATE SOURCE` statement is issued. This means that the computed start offsets will be the **same** for all views depending on the source and **stable** across restarts.
+
+If you need to limit the amount of data maintained as state after source creation, consider using [temporal filters](/guides/temporal-filters/) instead.
```

Comment:
@andrioni, how does this sound for #9611?

Reply:
LGTM!